src/graphtypes/simplegraphs/simpledigraph.jl

```diff
@@ -16,12 +16,8 @@ eltype(x::SimpleDiGraph{T}) where T = T
 
 # DiGraph{UInt8}(6), DiGraph{Int16}(7), DiGraph{Int8}()
 function (::Type{SimpleDiGraph{T}})(n::Integer = 0) where T<:Integer
-    fadjlist = Vector{Vector{T}}()
-    badjlist = Vector{Vector{T}}()
-    for _ = one(T):n
-        push!(badjlist, Vector{T}())
-        push!(fadjlist, Vector{T}())
-    end
+    fadjlist = [Vector{T}() for _ = one(T):n]
```

Comment:
Just curious, will this `one(T)` go away in Julia v0.7 and beyond?

Reply:
I don't know. Was there talk of deprecating it?

Reply:
I don't' know, I was just curious why we need `one(T)` in this range expression when it doesn't participate in the `Vector{T}()` part. Should the argument to the function be `n::T` instead or you are allowing different integer types on purpose here for type coercion?

Reply:
We don't necessarily need `one(T)` here. I loosened `n` precisely for type coercion.